### PR TITLE
Update automated commit status comment

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 5e0caa669e0c8e9dd6494f45289c90f75f4e419a with description of existing statuses. It's updated for the latest CI running*
📄 [Click here](https://s3.amazonaws.com/clickhouse-test-reports/54441/5e0caa669e0c8e9dd6494f45289c90f75f4e419a/ci_running.html) to open a full report in a separate page
The overall status of the commit is 🟡 pending

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tbody>
</table>
<details><summary>Successfull checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
<tbody>
</table>
</details>

**Comment by alexey-milovidov** [comment]
Wow, it works! Wouldn't it be better if we put the status symbol in front of the title?
(Easier to scan the list of pull requests).

**Comment by vdimir** [comment]
@alexey-milovidov 

Checked both options editing html and setting random badges.
Badge in the beginning indeed looks better imo.

<details>
<summary>Screenshots</summary>


Screenshot 2023-09-11 at 12 42 10

---

Screenshot 2023-09-11 at 12 42 20


</details>

File: tests/ci/commit_status_helper.py [hunk 1]
```diff
@@ -255,7 +295,14 @@ def create_ci_report(pr_info: PRInfo, statuses: CommitStatuses) -> str:
         log_urls = None
         if status.target_url is not None:
             log_urls = [status.target_url]
-        test_results.append(TestResult(status.context, status.state, log_urls=log_urls))
+        raw_logs = None
+        if status.description:
+            raw_logs = status.description
+        test_results.append(
+            TestResult(
+                status.context, status.state, log_urls=log_urls, raw_logs=raw_logs
```
Comment (vdimir) [inline tests/ci/commit_status_helper.py:303]: Don't work yet (no expandable descriptions here https://s3.amazonaws.com/clickhouse-test-reports/54441/6a51d9fd7d2a373741f79ac7f9100225d79ef8bd/ci_running.html), going to check what's wrong

**Comment by Felixoid** [comment]
Won't it be TOO annoying to have these strings for every update?..

image

**Comment by vdimir** [comment]
> The mailboxes will have up to 4 different threads (original, and 3 with different statuses)

Let's check it

**Comment by vdimir** [comment]
bump

**Comment by Felixoid** [comment]
image

It looks strange..

**Comment by Felixoid** [comment]
@alexey-milovidov Vladimir and came to a model where we'd use only two statuses, red and green, and change them only back and forth as labels `pr-status-🟢` and `pr-status-🔴`

Changing labels are sent to all subscribers, so it's the best balance we could think of.

File: tests/ci/commit_status_helper.py [hunk 2]
```diff
@@ -255,7 +295,14 @@ def create_ci_report(pr_info: PRInfo, statuses: CommitStatuses) -> str:
         log_urls = None
         if status.target_url is not None:
             log_urls = [status.target_url]
-        test_results.append(TestResult(status.context, status.state, log_urls=log_urls))
+        raw_logs = None
+        if status.description:
+            raw_logs = status.description
```
Comment (Felixoid) [inline tests/ci/commit_status_helper.py:300]:
```suggestion
        raw_logs = status.description or None
```

**Comment by Felixoid** [comment]
image

Just in case ¯\\\_(ツ)_/¯

**Comment by alexey-milovidov** [comment]
Thanks, it looks like a good idea!

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 3]
```diff
@@ -7656,8 +7656,7 @@ void StorageReplicatedMergeTree::replacePartitionFrom(
             entry_replace.columns_version = -1;
         }
 
-        if (replace)
-        {
```
Comment (Felixoid) [inline src/Storages/StorageReplicatedMergeTree.cpp:7660]: Looks good. Let's revert it and run full checks to see the other reports

**Review by Felixoid — CHANGES_REQUESTED** [review]

File: src/Storages/StorageReplicatedMergeTree.cpp (hunk 3 above)
Comment (Felixoid) [reply src/Storages/StorageReplicatedMergeTree.cpp:7660]: And there's a lost remaining in report.py: `if (classes.contains("failed-content") || classes.contains("failed-content.open"))`

**Review by Felixoid — APPROVED** [review]
Schitts Creek Perfect

File: tests/ci/commit_status_helper.py [hunk 4]
```diff
@@ -255,7 +292,12 @@ def create_ci_report(pr_info: PRInfo, statuses: CommitStatuses) -> str:
         log_urls = None
         if status.target_url is not None:
             log_urls = [status.target_url]
```
[the remark this answers is not in the document]
Comment (Felixoid) [reply tests/ci/commit_status_helper.py]: Nice, it's fixed! https://s3.amazonaws.com/clickhouse-test-reports/54441/acb559ce8bd385d311c27aabb3e88fbf4a1cfa22/ci_running.html

**Review by Felixoid — APPROVED** [review]

**Comment by Felixoid** [comment]
https://github.com/ClickHouse/ClickHouse/pull/54551 will address test_keeper_reconfig_remove_many/test.py::test_reconfig_remove_2_and_leader

**Pull request merged**